tensorflow/core/kernels/transpose_op.cc

```diff
@@ -133,7 +133,17 @@ void TransposeOp::Compute(OpKernelContext* ctx) {
   OP_REQUIRES(ctx, TensorShapeUtils::IsVector(perm.shape()),
               errors::InvalidArgument("perm must be rank 1, got shape ",
                                       perm.shape().DebugString()));
-
+  // Validation for negative values inside `perm`.
+  auto perm_vector = perm.vec<int>();
+  auto perm_size = perm.NumElements();
+  for (int i = 0; i < perm_size; ++i) {
+    if (perm_vector(i) < 0) {
+      return absl::InvalidArgumentError(absl::StrCat(
```

Comment:
You cannot use `return` here. The function is `void`. You have to use the `OP_REQUIRES` macro instead

Reply:
Thanks for correction.Done the changes as required.